packages/web/components/swap-tool/price-selector.tsx

```diff
@@ -0,0 +1,490 @@
+import { Menu, Transition } from "@headlessui/react";
+import { Dec, PricePretty } from "@keplr-wallet/unit";
+import { DEFAULT_VS_CURRENCY, MaybeUserAssetCoin } from "@osmosis-labs/server";
+import { Asset } from "@osmosis-labs/types";
+import { getAssetFromAssetList } from "@osmosis-labs/utils";
+import classNames from "classnames";
+import { observer } from "mobx-react-lite";
+import Image from "next/image";
+import { parseAsBoolean, parseAsString, useQueryState } from "nuqs";
+import React, { Fragment, memo, useEffect, useMemo } from "react";
+
+import { Icon } from "~/components/assets";
+import { AssetLists } from "~/config/generated/asset-lists";
+import { useDisclosure, useTranslation } from "~/hooks";
+import { useOrderbookSelectableDenoms } from "~/hooks/limit-orders/use-orderbook";
+import { AddFundsModal } from "~/modals/add-funds";
+import { useStore } from "~/stores";
+import { formatFiatPrice } from "~/utils/formatter";
+import { api } from "~/utils/trpc";
+
+type AssetWithBalance = Asset & MaybeUserAssetCoin;
+
+const UI_DEFAULT_QUOTES = ["USDC", "USDT"];
+
+const VALID_QUOTES = [
```

Comment:
Are these denoms available in our asset list? If so, we can use `MainnetAssetSymbols` types to make sure these denoms are valid during build.